mm/iob/iob_free.c

```diff
@@ -112,6 +115,15 @@ FAR struct iob_s *iob_free(FAR struct iob_s *iob)
               next, next->io_pktlen, next->io_len);
     }
 
+#ifdef CONFIG_IOB_ALLOC
+  if (iob->io_free != NULL)
+    {
+      iob->io_free(iob->io_data);
+      kmm_free(iob);
```

Comment:
```suggestion
      iob->io_free(iob);
```

Reply:
@anchao we need to free iob head when driver call iob_alloc_with_data with custom free callback.